lib/python/treadmill_aws/aws.py

```diff
@@ -50,3 +52,36 @@ def int2str(number, base=16, symbol=None):
     basestr = ('-' if negative else '') + symbol[remainder] + basestr
 
     return basestr
+
+
+def str2int(word, base=16, symbol=None):
+    """Convert a alphanumerical string in a given base to base 10 integer
+    """
+    if not symbol:
+        symbol = '0123456789abcdefghijklmnopqrstuvwxyz'
+    if len(symbol) < 2:
+        raise ValueError('Length of symbol list should be >= 2')
+    if base < 2 or base > len(symbol):
+        raise ValueError('Base must be in range 2-{}'.format(len(symbol)))
+
+    negative = True if word.startswith('-') else False
+    word.strip('-')
+
+    number = None
+    step = 0
+    for char in reversed(word):
+        if number is None:
+            number = symbol.index(char) * pow(base, step)
+        else:
+            number += symbol.index(char) * pow(base, step)
+        step += 1
+
+    return number * -1 if negative else number
+
+
+def hostname_datetime(hostname):
+    """Extract alphanumeric timestamp from hostname and return datetime object
+    """
+    # Note naming convention may change, rely on ldap object creation time
+    ts = str2int(hostname.split('.')[0].split('-').pop(), base=32)
```

Comment:
This logic should be broken down into several operations, not written as one line.  Not all hostnames will follow the same pattern- treadmill-node-{date} is just the default. 
You could fall back to querying LaunchTime from the EC2 instance?

Reply:
it is not being used (ldap timestamp is used in the code), but it is nice little utility function to have. i would keep it, just rename datetime_from_suffix or something like that. i think str2int is enough, and this one can be removed, and it is trivial.

Reply:
done